app/models/tag.rb

```diff
@@ -13,7 +13,7 @@ class Tag < ActiveRecord::Base
   has_many :tag_groups, through: :tag_group_memberships
 
   def self.tags_by_count_query(opts={})
-    q = TopicTag.joins(:tag, :topic).group("topic_tags.tag_id, tags.name").order('count_all DESC')
+    q = Topic.unscoped { TopicTag.joins(:tag, :topic).group("topic_tags.tag_id, tags.name").order('count_all DESC') }
```

Comment:
It turns out this case should be handled, but `counter_caches` rarely work.

```
    unused_tags = DiscourseTagging.filter_allowed_tags(
      Tag.where(topic_count: 0),
      guardian,
      { for_input: params[:filterForInput], term: params[:q], category: category, selected_tags: params[:selected_tags] }
    )
```

Tags on deleted topics still have a topic_count of 1. I want to remove the topic_count column from tags.

When a topic is deleted, maybe it should lose its tags. Thoughts?